agent/server_test.go

```diff
@@ -511,6 +648,14 @@ func testDialRPC(port int, certFile string) dialRPC {
 	}
 }
 
+func tags(tags ...string) map[string]string {
+	result := map[string]string{}
+	for i := 0; i < len(tags); i += 2 {
+		result[tags[i]] = tags[i+1]
```

Comment:
You'll get an "index out of bounds" panic here if somebody passes an odd number of tags.